### ISSUE TITLE
should read nested components fixtures

### ISSUE THREAD

**Issue by amrdraz** [react-cosmos/react-cosmos#382]
It seems to me that no fixture files are read beyond the first component level.

I have a project architecture that looks like this
<img width="288" alt="screen shot 2017-06-30 at 00 26 55" src="https://user-images.githubusercontent.com/1708559/27713221-dda7ed62-5d2a-11e7-8492-41ef4c836bfb.png">
But while cosmos can detect the components in the project the playground looks like this
<img width="383" alt="screen shot 2017-06-30 at 00 24 32" src="https://user-images.githubusercontent.com/1708559/27713191-b3014e28-5d2a-11e7-93c8-c189873065ad.png">

it seems to me that the fixture files are not included

I tried modifying the cosmos.config.js file to no avail

```
const glob = require('glob')
module.exports = {
  componentPaths: ['src/components'],
  // componentPaths: glob.sync('src/{,**/}components'),
  // getComponentName: path => path.match(/src\/components\/(?:.+\/components\/)(.+)\.jsx?$/),
  fixturePaths: glob.sync('src/{,**/}components/{,**/}__fixtures__/'),
  ...
}
```

When I tried `fixturePaths: glob.sync('src/{,**/}components'),` I got index.js (the component index not the fixtures)

**Comment by skidding** [comment]
Hey @amrdraz. Can you please try this?

```
components/Board/index.js
components/Board/__fixtures__/initial.js
components/Board/components/BoardSquare/index.js
components/Board/components/BoardSquare/__fixtures__/even.js
components/Board/components/BoardSquare/__fixtures__/odd.js
components/Board/components/BoardSquare/components/Square/index.js
components/Board/components/BoardSquare/components/Square/__fixtures__/foo.js
components/Board/components/Knight/index.js
components/Board/components/Knight/__fixtures__/bar.js
```

**Comment by amrdraz** [comment]
Greate that works, (for some reason I couldn't start the project so I started a new one)

But I found a small problem

It seems as though all files under __fixtures__ in the same component get loaded, so in this example the slide fixture is being read twice under `BannerCarousel`

<img width="284" alt="screen shot 2017-07-21 at 13 37 47" src="https://user-images.githubusercontent.com/1708559/28462098-f8195406-6e19-11e7-9c29-d9ccf8f53c30.png">

Inspecting the code I realized the reason is that `getMatchingFixtures`matches the `componentName` against all available fixture paths

and since the nested component by definition included the parent component
the nested component fixture appear in the parent component fixtures.

```
For componentName:"BannerCarousel/components/Slide" fixtures paths match
false '/Users/draz/web/cairousel-web/src/components/__fixtures__/Text/header.js'
false '/Users/draz/web/cairousel-web/src/components/BannerCarousel/__fixtures__/default.js'
slide /Users/draz/web/cairousel-web/src/components/BannerCarousel/components/Slide/__fixtures__/slide.js
false '/Users/draz/web/cairousel-web/src/components/Button/__fixtures__/default.js'

For componentName:"BannerCarousel" fixtures paths match
false '/Users/draz/web/cairousel-web/src/components/__fixtures__/Text/header.js'
default /Users/draz/web/cairousel-web/src/components/BannerCarousel/__fixtures__/default.js
slide /Users/draz/web/cairousel-web/src/components/BannerCarousel/components/Slide/__fixtures__/slide.js
false '/Users/draz/web/cairousel-web/src/components/Button/__fixtures__/default.js'
```
The culprit is the second regular expression in 

```js
const buildPathMatchers = (componentName, fixturesDir) => [
  `/${fixturesDir}/${componentName}/([^/]+)\\.(${FIXTURE_EXTENSIONS_REGEX})$`,
  `/${componentName}/(?:.+/)?${fixturesDir}/([^/]+)\\.(${FIXTURE_EXTENSIONS_REGEX})$`,
  `/${componentName}/([^/]+)\\.(${FIXTURE_EXTENSIONS_REGEX})$`,
];
```

I removed `/(?:.+/)?` from the second line which I believe more strictly matches the behavior that is expected 

```js
const buildPathMatchers = (componentName, fixturesDir) => [
  `/${fixturesDir}/${componentName}/([^/]+)\\.(${FIXTURE_EXTENSIONS_REGEX})$`,
  - `/${componentName}/(?:.+/)?${fixturesDir}/([^/]+)\\.(${FIXTURE_EXTENSIONS_REGEX})$`,
  + `/${componentName}/${fixturesDir}/([^/]+)\\.(${FIXTURE_EXTENSIONS_REGEX})$`,
  `/${componentName}/([^/]+)\\.(${FIXTURE_EXTENSIONS_REGEX})$`,
];
```

And everything works fine, I forked the repo and ran the tests after my modification and everything checks out.

should I create a pull request?

**Comment by amrdraz** [comment]
ok so the `separate-packages` test fails because getComponentName when defined in the config file assumes that the __fixture__ folder must lie under the component in general, but as pointed out by my issue  this behavior would leade to the component matching with it's nested component's fixtures.

I believe the default behavior should be to match the componentName and only 
consider fixtures as shown in the docs (ie not to match Foo/src/__fixtures as a default)

something like capturing nested path should be handled by replacing the `getMatchingFixtures` which is already something that's possible in the config file

**Comment by amrdraz** [comment]
I updated the test case to reflect the argument and added another case for fixtures that are nested under components with fixtures which will fail under the old way the regular expression was setup.

now all tests pass

**Comment by skidding** [comment]
Sounds great. Can't wait to review the PR on Monday. It might also fix #387

**Comment by skidding** [comment]
Fixed by #398
